gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -175,6 +188,135 @@ var fluid = fluid || require("infusion"),
         return togo;
     };
 
+    // true if one of the values matches across the two arrays (ignoring the indexing)
+    gpii.matchMakerFramework.utils.containsSameValue = function (arr1, arr2) {
+        var obj1 = {};
+        var i;
+        // convert arr1 to object for fast and easy comparison
+        for (i in arr1) {
+            obj1[arr1[i]] = true;
+        }
+        // compare:
+        for (i in arr2) {
+            if (arr2[i] in obj1) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    /**
+     */
+    gpii.matchMakerFramework.utils.disposeFromPriority = function (solrecs, data) {
+        // sort into prioritized map: 1024: [sola, solb], 200 [solc]
+        var prioritiesMap = {};
+        fluid.each(solrecs, function (sol) {
+            if (sol.priority) {
+                fluid.set(prioritiesMap, [sol.priority, sol.index], true);
+            }
+        });
+        // sort priority entries desc
+        var priorities = Object.keys(prioritiesMap).sort(function (a, b) { return b - a; });
```

Comment:
This needs to be an explicitly named comparator that the user can identify and override